services/authoring/src/repositories/apps-repository.ts

```diff
@@ -33,12 +40,81 @@ export default class AppsRepository {
     return this.apps;
   }
 
-  async saveApp(appId, manifest, author) {
-    return await this._gitTransactionManager.write(async repo => {
+  async createApp(appId: string, manifest: AppManifest, author) {
+    return await this._gitTransactionManager.write(async (repo) => {
       await repo.updateFile(`external_apps/${appId}.json`, JSON.stringify(manifest, null, 4));
       const commitId = await repo.commitAndPush(`created app ${appId}`, author);
       await this.refresh();
       return commitId;
     });
   }
+
+  async updateApp(appId: string, manifest: Partial<Pick<AppManifest, 'name' | 'permissions'>>, author) {
+    const appFileName = `external_apps/${appId}.json`;
+    return await this._gitTransactionManager.write(async (repo) => {
+      const app = this.apps[appId];
+      if (!app) {
+        throw new Error(`The app with id ${appId} doesn't exist`);
+      }
+      const updatedApp = { ...app, ...manifest };
+      await repo.updateFile(appFileName, JSON.stringify(updatedApp));
+      const commitId = await repo.commitAndPush(`updated app ${appId}`, author);
+      await this.refresh();
+      return commitId;
+    });
+  }
+
+  async deleteApp(appId: string, author) {
+    return await this._gitTransactionManager.write(async (repo) => {
+      await repo.deleteFile(`external_apps/${appId}.json`);
+      const commitId = await repo.commitAndPush(`deleted app ${appId}`, author);
+      await this.refresh();
+      return commitId;
+    });
+  }
+
+  async createSecretKey(appId: string, secretKey: AppSecretKey, author) {
+    const appFileName = `external_apps/${appId}.json`;
+    return await this._gitTransactionManager.write(async (repo) => {
+      const app = this.apps[appId];
+      if (!app) {
+        throw new Error(`The app with id ${appId} doesn't exist`);
+      }
+
+      if (app.secretKeys.some(k => k.id === secretKey.id)) {
+        throw new Error(`Key ${secretKey.id} already exists`);
+      }
+      app.secretKeys.push(secretKey);
+      await repo.updateFile(appFileName, JSON.stringify(app));
+      const commitId = await repo.commitAndPush(`created secret key ${secretKey.id} for app ${appId}`, author);
+      await this.refresh();
+      return commitId;
+    });
+  }
+
+  getSecretKeys(appId: string) {
+    return this.apps[appId].secretKeys.reduce((acc, cur, idx) => ({ ...acc, [`legacy-${idx}`]: cur }), {});
```

Comment:
Since we have ramda, we can use R.indexBy:
https://ramdajs.com/docs/#indexBy
A bit cleaner.